OpenXmlPowerTools/PtOpenXmlUtil.cs

```diff
@@ -5796,4 +6162,66 @@ public WorksheetAlreadyExistsException(string sheetName)
         {
         }
     }
+
+    public static class NamespaceAttributeUtil
+    {
+        public static XAttribute[] NamespaceAttributes;
+        public static XAttribute[] PowerToolsNamespaceAttributes;
+
+        private static XAttribute[] BaseNamespaceAttributes =
+        {
+            new XAttribute(XNamespace.Xmlns + "wpc", WPC.wpc),
+            new XAttribute(XNamespace.Xmlns + "mc", MC.mc),
+            new XAttribute(XNamespace.Xmlns + "o", O.o),
+            new XAttribute(XNamespace.Xmlns + "r", R.r),
+            new XAttribute(XNamespace.Xmlns + "m", M.m),
+            new XAttribute(XNamespace.Xmlns + "v", VML.vml),
+            new XAttribute(XNamespace.Xmlns + "wp14", WP14.wp14),
+            new XAttribute(XNamespace.Xmlns + "wp", WP.wp),
+            new XAttribute(XNamespace.Xmlns + "w10", W10.w10),
+            new XAttribute(XNamespace.Xmlns + "w", W.w),
+            new XAttribute(XNamespace.Xmlns + "w14", W14.w14),
+            new XAttribute(XNamespace.Xmlns + "w15", W15.w15),
+            new XAttribute(XNamespace.Xmlns + "w16cid", W16CID.w16cid),
+            new XAttribute(XNamespace.Xmlns + "w16se", W16SE.w16se),
+            new XAttribute(XNamespace.Xmlns + "wpg", WPG.wpg),
+            new XAttribute(XNamespace.Xmlns + "wpi", WPI.wpi),
+            new XAttribute(XNamespace.Xmlns + "wne", WNE.wne),
+            new XAttribute(XNamespace.Xmlns + "wps", WPS.wps),
+        };
+
+        private static XAttribute[] CommonNamespaceAttributes =
+        {
+            new XAttribute(MC.Ignorable, "w14 wp14 w15 w16se w16cid"),
+        };
+
+        private static XAttribute[] PowerToolsSpecificNamespaceAttributes =
+        {
+            new XAttribute(XNamespace.Xmlns + "pt", PtOpenXml.pt),
+            new XAttribute(MC.Ignorable, "w14 wp14 w15 w16se w16cid pt"),
+        };
+
+        static NamespaceAttributeUtil()
+        {
+            //building the most common attribute list
+            NamespaceAttributes =
+                new XAttribute[BaseNamespaceAttributes.Length + CommonNamespaceAttributes.Length];
+            BaseNamespaceAttributes.CopyTo(NamespaceAttributes, 0);
+            CommonNamespaceAttributes.CopyTo(NamespaceAttributes, BaseNamespaceAttributes.Length);
+
+            //and the Power Tools specific list
+            PowerToolsNamespaceAttributes =
+                new XAttribute[BaseNamespaceAttributes.Length + PowerToolsSpecificNamespaceAttributes.Length];
+            BaseNamespaceAttributes.CopyTo(PowerToolsNamespaceAttributes, 0);
+            PowerToolsSpecificNamespaceAttributes.CopyTo(PowerToolsNamespaceAttributes, BaseNamespaceAttributes.Length);
+        }
+
+        public static void ApplyNamespaceAttributes(XElement element)
+        {
+            if (element != null)
+                foreach (var attr in NamespaceAttributes)
+                    if (element.Attribute(attr.Name) == null)
+                        element.Add(attr);
+        }
```

Comment:
I believe that this is only required because your other changes did not go "all the way". Please see my comments on the `DocumentBuilder` and `HtmlToWmlConverter` above.